### DISCUSSION TITLE
.lazy.lua: how to set non-plugin options?

### DISCUSSION THREAD

**Discussion by meleu** [LazyVim/LazyVim#4920] [Q&A]
I'm working in different ruby projects. Therefore I'm using the `extras.lang.ruby`.

As [per documentation](https://www.lazyvim.org/extras/lang/ruby), I can configure the Ruby extra in my `lua/config/options.lua` like this:

```lua
vim.g.lazyvim_ruby_lsp = "ruby_lsp"
vim.g.lazyvim_ruby_formatter = "rubocop"
```

- The most "modern" ruby projects usually work fine with `ruby_lsp`+`standardrb`.
- The "legacy" ones work better with `solargraph`+`rubocop`.

Recently I learned we can have a `.lazy.lua` file in the project's directory for project specific configurations.

The problem is that apparently the `.lazy.lua` only accepts plugin specs.

Unfortunately I don't know lua enough to figure out how to put those `vim.g.*` configurations as they were part of a plugin spec.

Any help on this?

Thanks in advance. :heart:

**Comment by folke** [comment]
```lua
vim.g.lazyvim_ruby_lsp = "ruby_lsp"
vim.g.lazyvim_ruby_formatter = "rubocop"

return {}
```

**Reply by meleu** [reply]
That solution doesn't result in an error but also doesn't produce the desired result.

In my tests it seems like the project's `.lazy.lua` was ignored.

In this screenshot I'm showing the ruby configs of the two files (the global and the project's one):

![image](https://github.com/user-attachments/assets/79c678e2-ea4a-495c-a81b-84ad254be029)


And here I'm showing the output of `:ConformInfo` (showing that it's respecting only the global configuration, ignoring the one in `.lazy.lua`):

![image](https://github.com/user-attachments/assets/90433c1d-aeb3-412f-8fd7-f0bc5edc4434)

**Reply by folke** [reply]
sorry, I didnt fully read what you were trying to do.

`.lazy.lua` gets sourced after everything else, so that won't work for what you are trying to do.

You can easily use `vim.secure` in your `options.lua` to read local options yourself.

**Reply by moonfruit** [reply]
I have the same problem. Is there any solution for this? @meleu @folke

**Reply by dpetka2001** [reply]
@moonfruit I've taken the function that `lazy.nvim` uses (I borrowed it from there after all) and just changed a bit how data is handled. You can put this somewhere in your configuration and then call the function in your `options.lua` file
```lua
function M.local_global_vars()
  local path = vim.uv.cwd()
  while path and path ~= "" do
    local file = path .. "/.lazy.lua"
    if vim.fn.filereadable(file) == 1 then
      local data = vim.secure.read(file)
      if data then
        local lines = {}
        for line in data:gmatch("[^\r\n]+") do
          if line:match("^return%s*{") then
            break
          end
          table.insert(lines, line)
        end
        local code = table.concat(lines, "\n")
        loadstring(code, ".lazy.lua")()
        break
      end
    end
    local p = vim.fn.fnamemodify(path, ":h")
    if p == path then
      break
    end
    path = p
  end
end
```
